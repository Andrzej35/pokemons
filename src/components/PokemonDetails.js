import React from 'react'
import { graphql } from 'react-apollo'
import { Link } from 'react-router-dom'
import { getPokemonQuery } from '../queries/queries'

const PokemonDetails = (props) => {
    const { pokemon } = props.data
    if (!pokemon) {
        return <div>Loading pokemon's information...</div>
    }
    return (

        <div id="pokemon-details">

            <Link to={`/`}>Go back to Pokemon's list</Link>
            <h1>Pokemon's information</h1>
            <div className="pokemon-details">
                <img src={pokemon.image} alt={pokemon.name} />
                <h3>{pokemon.name}</h3>
                <p>{pokemon.number}</p>
                <p>{pokemon.types}</p>
                <p>{pokemon.resistant}</p>
                <p>{pokemon.weaknesses}</p>
            </div>

        </div>
    )
}

export default graphql(getPokemonQuery, {
    options: props => {
        return {
            variables: {
                id: props.match.params.id
            }
        }
    }
})(PokemonDetails)
