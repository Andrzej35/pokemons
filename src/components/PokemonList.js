import React, { Component } from 'react'
import { graphql } from 'react-apollo'
import { getPokemonsQuery } from '../queries/queries'
import { Link } from 'react-router-dom'

class PokemonList extends Component {

    displayPokemons() {
        const { data } = this.props
        if (data.loading) {
            return (<div>loading pokemons ...</div>)
        } else {
            // console.log(data.pokemons.map(pokemon => pokemon))
            return data.pokemons.map(pokemon => {
                return (<li key={pokemon.id} onClick={(e) => { this.setState({ selected: pokemon.id, name: pokemon.name }) }}>
                    <Link to={`/pokemon/${pokemon.id}`}>
                        <div className="list">
                            <div className="image"><img className="item-image" src={pokemon.image} alt={pokemon.name} width="100" /></div>
                            <div className="content"><h3><strong>Name: </strong>{pokemon.name}</h3>
                                <p><strong>Number: </strong>{pokemon.number}</p>
                            </div>
                        </div>
                    </Link>
                </li>)
            })
        }
    }
    render() {
        console.log(this.state)
        return (
            <div>
                <h1>Pokemons List:</h1>
                <ul className="pokemon-list">
                    {this.displayPokemons()}
                </ul>
            </div>
        )
    }
}

export default graphql(getPokemonsQuery)(PokemonList)
