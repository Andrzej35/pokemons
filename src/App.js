import React from 'react'
import ApolloClient from 'apollo-boost'
import { ApolloProvider } from 'react-apollo'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'

import PokemonList from './components/PokemonList'
import PokemonDetails from './components/PokemonDetails'
import Error404 from './components/Error404'

const client = new ApolloClient({
  uri: 'https://graphql-pokemon.now.sh/graphql'
})



const App = () => {
  return (
    <ApolloProvider client={client}>
      <div className="App">
        <Router>
          <Switch>
            <Route path="/" exact component={PokemonList} />
            <Route path="/pokemon/:id" exact component={PokemonDetails} />
            <Route component={Error404} />
          </Switch>
        </Router>
      </div>
    </ApolloProvider>
  );
}

export default App
