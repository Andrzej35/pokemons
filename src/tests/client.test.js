import renderer, { create } from 'react-test-renderer'
import React from 'react'
import { shallow, mount, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import App from '../App'
import PokemonList from '../components/PokemonList'
import PokemonDetails from '../components/PokemonDetails'



configure({ adapter: new Adapter() })

describe('[<App />]', () => {
    it('[<App />] shows pokemons list', () => {
        const component = renderer.create(<App />)
        const tree = component.toJSON()
        component.update()
        expect(tree).toMatchSnapshot()
    })

    it('[<App />] should have heading 1', () => {
        const wrap = mount(<App />)
        expect(wrap.find('h1')).toHaveLength(1)
    })

    it('[<App />] should render correct text in heading', () => {
        const wrap = shallow(<h1>Pokemon's List:</h1>)
        expect(wrap.text()).toEqual('Pokemon\'s List:')
    })
})

describe('[<PokemonList />]', () => {
    it('[<PokemonList />] renders correctly', () => {
        const wrapper = shallow(<PokemonList />)
        const fetched = true

        const pkmns = [
            { name: "bulbasaur" },
            { name: "ivysaur" },
            { name: "venusaur" }
        ];

        wrapper.setState({
            fetched,
            pkmns
        })

        expect(toJson(wrapper)).toMatchSnapshot()
    });
})
