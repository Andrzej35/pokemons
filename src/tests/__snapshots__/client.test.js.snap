// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<App />] [<App />] shows pokemons list 1`] = `
<div
  className="App"
>
  <div>
    <h1>
      Pokemon's List:
    </h1>
    <ul
      className="pokemon-list"
    >
      <div>
        loading pokemons ...
      </div>
    </ul>
  </div>
</div>
`;

exports[`[<PokemonList />] [<PokemonList />] renders correctly 1`] = `
<Query
  displayName="Apollo(PokemonList)"
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": undefined,
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "first",
                    },
                    "value": Object {
                      "kind": "IntValue",
                      "value": "151",
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "pokemons",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "image",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "number",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "name",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 91,
        "start": 0,
      },
    }
  }
  skip={false}
>
  <Component />
</Query>
`;
